Neos.Utility.Schema/Classes/TYPO3/Flow/Utility/SchemaValidator.php

```diff
@@ -624,4 +625,50 @@ protected function isDictionary(array $phpArray)
     {
         return array_keys($phpArray) !== range(0, count($phpArray) - 1);
     }
+    
+    /**
+     * Generate a short description for the given schema to be used in error-messages
+     *
+     * @param $schema
+     * @return string
+     */
+    protected function getTypeInfo($type) {
+        if (is_scalar($type)) {
+            return (string)$type;
+        } else if (is_array($type)){
+            if ($this->isSchema($type) && array_key_exists('type', $type)){
+                return $this->getTypeInfo($type['type']);
+            } else if ($this->isNumericallyIndexedArray($type)) {
+                $subTypeInfos = [];
+                foreach ($type as $subType) {
+                    $subTypeInfos[] = $this->getTypeInfo($subType);
+                }
+                return '[' . implode(', ', $subTypeInfos) . ']';
+            } else {
+                return 'dictionary';
+            }
+        } else {
+            return gettype($type);
+        }
```

Comment:
could be simplified..

    if (is_scalar($type)) {
        return (string)$type;
    }
    if (is_array($type)) {
        if ($this->isSchema($type) && array_key_exists('type', $type)) {
           return $this->getTypeInfo($type['type']);
        }
        if ($this->isNumericallyIndexedArray($type)) {
           $subTypeInfos = [];
           foreach ($type as $subType) {
               $subTypeInfos[] = $this->getTypeInfo($subType);
           }
           return '[' . implode(', ', $subTypeInfos) . ']';
       }
       return 'dictionary';
   }
   return gettype($type);